nil/internal/execution/state.go

```diff
@@ -213,17 +212,21 @@ func NewEVMBlockContext(es *ExecutionState) (*vm.BlockContext, error) {
 
 	currentBlockId := uint64(0)
 	var header *types.Block
+	time := uint64(0)
 	if err == nil {
 		header = data.Block()
 		currentBlockId = header.Id.Uint64() + 1
+		// TODO: we need to use header.Timestamp instead of but it's always zero for now.
```

Comment:
we need to either exclude it from hashed fields and make non-zero or completely remove from the block